hera_mc/cm_health.py

```diff
@@ -0,0 +1,133 @@
+# -*- mode: python; coding: utf-8 -*-
+# Copyright 2018 the HERA Collaboration
+# Licensed under the 2-clause BSD license.
+
+"""Functions to handle various cm table health checks.
+
+"""
+
+from __future__ import absolute_import, print_function
+import warnings
+from hera_mc import mc, cm_utils
+from hera_mc import part_connect as PC
+from hera_mc import cm_revisions
+
+
+class RevisionError(Exception):
+    def __init__(self, hpn):
+        # Call the base class constructor with the parameters it needs
+        message = "Multiple revisions found on {}".format(hpn)
+        super(RevisionError, self).__init__(message)
+
+
+class Connections:
+    def __init__(self, session=None):
+        if session is None:
+            db = mc.connect_to_mc_db(None)
+            self.session = db.sessionmaker()
+        else:
+            self.session = session
+        self.conndict = None
+
+    def get_conndict(self, session=None):
+        self.conndict = {}
+        self.multiples = []
```

Comment:
This should be a `set`, rather than a list, to make the `not in` check below more efficient and more precisely capture what this data structure is doing.